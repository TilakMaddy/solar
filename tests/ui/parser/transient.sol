contract C {
    uint256 transient;

    function f() external {
        transient = 1;
    }
}

contract D {
    uint256 transient transient;

    function f() external {
        transient = 2;
    }
}

contract E {
    function f(uint256 transient) external { 
        transient = 3;
    }
    
    function f2(uint256 transient, bool) external { 
        transient = 4;
    }

    function g(uint256 transient transient) external { //~ ERROR: data location can only be specified for array, struct or mapping types
        transient = 5;
    }

    function g2(uint256[] transient transient) external { //~ ERROR: invalid data location
        transient = 5;
    }

  }
