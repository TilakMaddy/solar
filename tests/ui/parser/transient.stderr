error: no visibility specified
  --> ROOT/tests/ui/parser/transient.sol:LL:CC
   |
LL |     function f(uint256 transient) {
   |              ^
   |
   = help: do you intend to add public?

error: no visibility specified
  --> ROOT/tests/ui/parser/transient.sol:LL:CC
   |
LL |     function f2(uint256 transient, bool) {
   |              ^^
   |
   = help: do you intend to add public?

error: no visibility specified
  --> ROOT/tests/ui/parser/transient.sol:LL:CC
   |
LL |     function g(uint256 transient transient) {
   |              ^
   |
   = help: do you intend to add public?

error: no visibility specified
  --> ROOT/tests/ui/parser/transient.sol:LL:CC
   |
LL |     function g2(uint256[] transient transient) {
   |              ^^
   |
   = help: do you intend to add public?

error: aborting due to 4 previous errors

